media/Config/pathAutoComplete.js

```diff
@@ -0,0 +1,118 @@
+  // because os maybe win32, divisor can be '\\'
+  let divisor = '/';
+  if (input.includes('\\')) {
+    divisor = '\\';
+  }
+  let paths = input.split(divisor);
+  let filename = paths[paths.length - 1].split('.');
+  let result = '';
+  switch (tool.type) {
+    case 'one-optimize': {
+      filename.splice(filename.length - 1, 0, 'opt');
+      paths[paths.length - 1] = filename.join('.');
+      result = paths.join(divisor);
+      break;
+    }
+    case 'one-quantize': {
+      if (filename.includes('opt')) {
+        filename[filename.indexOf('opt')] = 'quantized';
+      } else {
+        filename.splice(filename.length - 1, 0, 'quantized');
+      }
+      paths[paths.length - 1] = filename.join('.');
+      result = paths.join(divisor);
+      break;
+    }
+    case 'one-pack': {
+      while (filename.length > 1) {
+        filename.splice(1, 1);
+      }
+      filename[0] += '_pack';
+      paths[paths.length - 1] = filename.join('.');
+      result = paths.join(divisor);
+      break;
+    }
+    default: {
+      filename[filename.length - 1] = 'circle';
+      paths[paths.length - 1] = filename.join('.');
+      result = paths.join(divisor);
+    }
+  }
+  return result;
+};
+
+const setOutputPath = function(tool, newOutputPath) {
+  for (let i = 0; i < tool.options.length; i++) {
+    if (tool.options[i].optionName === 'output_path') {
+      tool.options[i].optionValue = newOutputPath;
+      break;
+    }
+  }
+};
+
+const outputToInput = function(toolIndex, nextInputValue) {
+  for (let k = toolIndex + 1; k < oneToolList.length; k++) {
+    if (oneToolList[k].use === true) {
+      for (let l = 0; l < oneToolList[k].options.length; l++) {
+        if (oneToolList[k].options[l].optionName === 'input_path') {
+          oneToolList[k].options[l].optionValue = nextInputValue;
+          break;
+        }
+      }
+      break;
+    }
+  }
+};
+
+
+
+// autoCompletePath make output_path based on input_path and copy former output_path to next
+// input_path for example, in import if input_path is 'filename.pb', then output_path will be
+// 'filename.circle' automatically and then in optimize output_path will be 'filename.opt.circle`
+// automatically
+// oneToolList come from tools.js it look like [oneImportBcq, oneImportOnnx, ..., oneProfile]
+// each tool has options for ONE-build.
+const autoCompletePath = function(tool) {
+  // tool argument decides which location to start
+  const index = oneToolList.indexOf(tool);
+  for (let i = index; i < oneToolList.length; i++) {
+    if (oneToolList[i].use === true) {
+      const inputPath = getInputPath(oneToolList[i]);
+      let outputPath = '';
+      if (inputPath.trim() !== '') {
+        outputPath = makeOutputPath(oneToolList[i], inputPath);
+        setOutputPath(oneToolList[i], outputPath);
+      } else {
+        outputPath = getOutputPath(oneToolList[i]);
+      }
+      if (outputPath.trim() !== '') {
+        outputToInput(i, outputPath);
+      }
```

Comment:
Looks much better to me as variable and function names express what it does and follow the logic by high(?) level :)
Thank you very much!